### DISCUSSION TITLE
milankovitch cycles?

### DISCUSSION THREAD

**Discussion by bputersznit** [Stellarium/stellarium#2807] [General]
I noticed there is no functionality for Milankovitch Cycles within Stellarium. Is this because it is BUILT IN and not mentioned in the User Guide? I DO see that Milkankovitch's minor work referenced below but nothing else. PLEASE ADVISE. BTW if you are not aware of his cycles pleas follow these links:

NASA ARTICLE
https://climate.nasa.gov/news/2948/milankovitch-orbital-cycles-and-their-role-in-earths-climate/

VIDEO
Dan Britt - Orbits and Ice Ages: The History of Climate
https://youtu.be/Yze1YAz_LYM

MILANKOVITCH SOLE MENTIONE IN IN STELLARIUM'S GUIDE:
"188 Chapter 14. Object Catalog Plugins proleptic, but the Romans did not keep the leap years commanded by JULIUS CAESAR until AUGUSTUS put things back in order. This means, displayed dates before 8 A.D. may be off by up to 3 days from historical accounts written by contemporaries. Gregorian This implementation acts like Stellarium with respect to year counting and counts signed negative and positive years, with a year zero between them. It shows dates in a Proleptic Gregorian calendar for dates before October 15, 1582. Given its improved rules for leap years which provide a mean length of the year of 365.2425 days, it keeps the seasons’ beginnings closer to the commonly known dates, at least for many more centuries in the past than the Julian calendar commonly used by historians. Revised Julian Calendar (also named Milankovi ́c Calendar) In 1923, the Serbian scientist MI- LUTIN MILANKOVI  ́C (1879–1958) proposed a calendar which should overcome the then 13-day calendar gap between the Eastern European Orthodox churches who still adhered to the Julian calendar and the rest of the world which followed the Gregorian calendar. It amends the 4-year Julian leap year cycle by omitting century years except for those where division by 900 leaves a remainder of 200 or 600. Therefore the mean length of the year is 365.242  ̄2 days, 24 seconds less than the Gregorian and within 2 seconds of the correct length of the mean tropical year. For a synchronisation with the Gregorian, October 1-13 1923 were omitted. Between March 1600 and February 2800 the calendar dates are identical to those in the Gregorian calendar. The calendar was adopted by several but not all Eastern Orthodox churches, although date of Easter is still computed according to the Julian calendar40. In historical context giving dates in the Revised Julian calendar for years before the dates of religious festivals were defined in the calendar makes no sense. Its idea was to have a continuous calendar as it was in use at the concile of Nicaea in A.D.325. Therefore dates before A.D.325 are displayed like in the traditional Julian calendar"

**Comment by bputersznit** [comment]
HMMM - It just occurred to me that I HAVE seen the procession of the rotation, so Milankovitch's cycle SHOULD be mostly covered.  What may NOT be covered is the Planets variation of magnitude due to Earth's (and other Planet's) orbital changes. These would be their SMA, Perihelion (& Aphelion), Tilt of Plane, etc. Sorry if it all has been accounted for. If it has PLEASE point this out to me, if it is in the code base.

**Reply by gzotti** [reply]
Stellarium's planet positions are from VSOP87, based on NASA/JPL DE200. Recommended use: -4000...+8000. 
Extended time range with NASA/JPL DE431/441, time range -13000...+17000. 
What is the period of Milankovich cycles again?

**Comment by bputersznit** [comment]
The following article explains them better than I can do:
https://en.wikipedia.org/wiki/Milankovitch_cycles
As far as I can remember any orbit is described by 6 numbers.The MCs mess with most (ALL?) of them.depending on the relative masses these parameters change.https://en.wikipedia.org/wiki/Orbital_elements

I vaguely remember the Keplerian elements: These are probable the most convenient for your calculations.There are others, vector based. could be better.Please compare to your code base.

Jupiter doesn't move much. 
Ceres is probably like a ping-pong ball.

Regards,Bernard P. 

    On Tuesday, November 1, 2022 at 05:33:40 PM EDT, Georg Zotti ***@***.***> wrote:  
 
 


Stellarium's planet positions are from VSOP87, based on NASA/JPL DE200. Recommended use: -4000...+8000.
Extended time range with NASA/JPL DE431/441, time range -13000...+17000.
What is the period of Milankovich cycles again?

**Reply by gzotti** [reply]
The question was rhetoric. How should 41.000 years or 100.000 years cycles be visualized in a 30.000 years fraction of the shortest cycle? Stellarium and all other desktop planetaria that I am aware of do not visualize this. More on Stellarium's limitations can be found in Appendix F.

**Comment by bputersznit** [comment]
so the simple answer is No?

Regards,Bernard P. 

    On Tuesday, November 1, 2022 at 06:11:21 PM EDT, Georg Zotti ***@***.***> wrote:  
 
 


The question was rhetoric. How should 41.000 years or 100.000 years cycles be visualized in a 30.000 years fraction of the shortest cycle? Stellarium and all other desktop planetaria that I am aware of do not visualize this. More on Stellarium's limitations can be found in Appendix F.

**Reply by gzotti** [reply]
If you want, Stellarium includes "Milanko" but not "vich". The covered time is just too short for a full cycle. You will also not detect the recurring phases of Venus in its 8-year cycle if you only observe for 5 years. However, the positions of the planets in the time covered are highly accurate. The MC is visible if you plot the osculating orbital elements. However, we don't provide osculating elements, just positions.

**Comment by bputersznit** [comment]
I wanted to test the night time skies back a few MYA - silly  me.
I guess Stellarium goes with a 'table based algorithm'I suspect a 'real time dynamic' set of equations would introduce ever increasing errors due to round off errors.Would probably need some 'infinite digit algorithm' akin to what MathCad does.It just creates higher digit variables (unlike floats) and never has to lose any precision.
https://en.wikipedia.org/wiki/MathcadRegards,Bernard P. 

    On Tuesday, November 1, 2022 at 09:12:44 PM EDT, Georg Zotti ***@***.***> wrote:  
 
 


If you want, Stellarium includes "Milanko" but not "vich". The covered time is just too short for a full cycle. You will also not detect the recurring phases of Venus in its 8-year cycle if you only observe for 5 years. However, the positions of the planets in the time covered are highly accurate. The MC is visible if you plot the osculating orbital elements. However, we don't provide osculating elements, just positions.
